pilot/pkg/networking/core/v1alpha3/listener_builder.go

```diff
@@ -477,44 +477,67 @@ func buildInboundCatchAllHTTPFilterChains(configgen *ConfigGeneratorImpl,
 			Bind:                       matchingIP,
 			InboundClusterName:         clusterName,
 		}
+		// Call plugins to install authn/authz policies.
+		var allChains []istionetworking.FilterChain
+		for _, p := range configgen.Plugins {
+			chains := p.OnInboundPassthroughFilterChains(in)
+			allChains = append(allChains, chains...)
+		}
+		if len(allChains) == 0 {
+			// Add one empty entry to the list if none of the plugins are interested in updating the filter chains.
+			allChains = []istionetworking.FilterChain{{}}
+		}
+
+		// Override the filter chain match to make sure the pass through filter chain captures the pass through traffic.
```

Comment:
this code is all duplicated in buildInboundCatchAllNetworkFilterChains, can we avoid this?

Reply:
The code is a little bit different in setting application protocols and listener protocols. We can extract some common code from these functions. The common code might be just a few lines( < 10 lines).